shared/actions/chat/attachment.js

```diff
@@ -94,96 +72,88 @@ function* _saveAttachment(conversationIDKey: Constants.ConversationIDKey, messag
   // download.
   if (endTime - startTime < 500) {
     for (let i = 0; i < 5; i++) {
-      yield put(Creators.downloadProgress(conversationIDKey, messageID, false, i + 1, 5))
+      yield put(Creators.downloadProgress(messageKey, false, (i + 1) / 5))
       yield delay(150)
     }
-    yield put(Creators.updateMessage(conversationIDKey, {downloadProgress: null}, messageID))
+    yield put(Creators.downloadProgress(messageKey, false, null))
   }
 
-  const downloadedMessage = yield select(Shared.messageSelector, conversationIDKey, messageID)
-  if (!downloadedMessage.downloadedPath) {
+  const {downloadedPath} = yield select(Constants.getLocalMessageStateFromMessageKey, messageKey)
+  if (!downloadedPath) {
     console.log('_saveAttachment: message failed to download!')
     return
   }
 
-  const destPath = yield call(downloadFilePath, downloadedMessage.filename)
+  const {filename} = yield select(Constants.getMessageFromMessageKey, messageKey)
```

Comment:
I'm not sure it's a good approach to get the filename out of the message data here. In theory, it could be out of date with respect to the original message data we fetched. In practice, I don't think we update filenames ever. Still, probably an antipattern. Alternative is to pass the filename into the action payload, and have the action creator take properties out of a message object?